src/main/java/com/powsybl/openloadflow/network/impl/LfBusImpl.java

```diff
@@ -42,6 +40,10 @@ public class LfBusImpl extends AbstractLfBus {
 
     private final List<String> bbsIds;
 
+    private final TopologyKind topologyKind;
+
+    private List<Integer> nodes = new ArrayList<>();
```

Comment:
should it be better to initialize in the constructor when the powsybl core bus is present 
something like : 
```java
if (topologyKind == TopologyKind.NODE_BREAKER) {
            nodes =  Arrays.stream(bus.getVoltageLevel().getNodeBreakerView().getNodes()).boxed().toList();
        }
```

Reply:
and juste get the node in getNodes()
```java
public List<Integer> getNodes() {
     return nodes;
}
```

Reply:
Because getNodes is called only in specific situations and a few times for a few buses, I would prefer to remove the field.
No mempry cost nor initialization costs in the majority of cases where it is not used.
Computed and allocated only when building a violation result in node breaker mode.